src/caret_analyze/architecture/architecture.py

```diff
@@ -501,13 +509,13 @@ def remove_variable_passing(self, node_name: str,
         callback_write: CallbackStructValue = \
             Util.find_one(lambda x: x.callback_name == callback_name_write, self.callbacks)
 
-        if callback_read.publish_topic_names:
+        if callback_read.publish_topics:
             context_reader = AssignContextReader(node)
-            for publish_topic_name in callback_read.publish_topic_names:
-                if callback_write.subscribe_topic_name:
-                    context_reader.remove_callback_chain(callback_write.subscribe_topic_name,
-                                                         publish_topic_name)
-
+            for publish_topic_name in callback_read.publish_topics:
+                if callback_write.subscribe_topic_name and publish_topic_name is not None:
+                    for pub_topic in publish_topic_name:
+                        context_reader.remove_callback_chain(callback_write.subscribe_topic_name,
+                                                             pub_topic.topic_name)
```

Comment:
Comparison of construction_order is necessary because this function should only remove callback chains related to `callback_name_write` and `callback_name_read`. I suggest changing these lines like below:
```
        if callback_read.publish_topics:
            context_reader = AssignContextReader(node)
            for publish_topic in callback_read.publish_topics:
                if callback_write.subscribe_topic_name and publish_topic_name is not None:
                    context_reader.remove_callback_chain(callback_write.subscribe_topic_name,
                                                         callback_write.subscription_construction_order,
                                                         publish_topic.topic_name,
                                                         publish_topic.construction_order)
```

Also, you need to add arguments `subscription_construction_order` and `publisher_construction_order` to remove_callback_chain() like below:
```
    def remove_callback_chain(
        self,
        subscribe_topic_name: str,
        subscription_construction_order: int,
        publish_topic_name: str,
        publisher_construction_order: int
    ) -> None:
        self._contexts = [
            context for context in self._contexts
            if (context['subscription_topic_name'],
                context['subscription_construction_order'] or 0,
                context['publisher_topic_name'],
                context['publisher_construction_order'] or 0,
                context['context_type']) !=
                (subscribe_topic_name, subscription_construction_order,
                publish_topic_name, publisher_construction_order, 'callback_chain')
        ]
```